sunpy/timeseries/timeseries_factory.py

```diff
@@ -537,6 +557,9 @@ def _apply_result(data_header_pairs, filepaths, result):
     if read:
         data_header_pairs.append(result)
     else:
+        if isinstance(result, io.IOBase):
+            warn("Falling back to using filepath", SunpyUserWarning)
+            result = result.name
```

Comment:
This should error.

Reply:
👍
is this error prompt appropiate?
```python
        if isinstance(result, io.IOBase):
            raise TypeError("file-handler not supported for unreadable files")
```

Reply:
Maybe a bit more info on what reader?

Reply:
uhh, I don't think its specifically attached to any reader; its triggered when the filetype is not supported(in the detection logic for timeseries)